### PR TITLE
ci: publish weekly metrics to slack

### PR DESCRIPTION
**Pull request by davejrt**
Push weekly ci metrics to slack.

Follow up: update action with new flags and create webhook for appropriate channel

## Test plan

Running locally and pushing report to #jh-bot-test

image

### CODE REVIEW

File: dev/buildchecker/main.go [hunk 1]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
...
 			}
+
 		}
+
 		if err := okayCli.Flush(); err != nil {
 			log.Fatal("Error posting to OKAYHQ okay.Flush: ", err.Error())
 		}
+
+	}
+
```
Comment (jhchabran) [inline dev/buildchecker/main.go:394]:
```suggestion
```

File: dev/buildchecker/main.go [hunk 2]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
 			if err != nil {
 				log.Fatal("Error storing OKAYHQ event okay.Push: ", err.Error())
 			}
+
 		}
+
 		if err := okayCli.Flush(); err != nil {
 			log.Fatal("Error posting to OKAYHQ okay.Flush: ", err.Error())
 		}
+
```
Comment (jhchabran) [inline dev/buildchecker/main.go:392]:
```suggestion
```

File: dev/buildchecker/main.go [hunk 3]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
...
+	}
+
+	if historyFlags.slackReportWebHook != "" {
+
```
Comment (jhchabran) [inline dev/buildchecker/main.go:396]:
```suggestion
```

File: dev/buildchecker/main.go [hunk 4]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
...
+		for _, incident := range incidents {
+			totalTime += incident
+		}
+
+		for _, flake := range flakes {
+			totalFlakes += flake
+		}
+
+		message := fmt.Sprintf(`
+		:bar_chart: Welcome to your weekly CI report for week ending: %s
```
Comment (jhchabran) [inline dev/buildchecker/main.go:414]: If we don't do this, we're prefixing the string with a newline, which is fine in the context of Slack, but can mess things up in other context. So it's a good habit to start straight after the backtick :) 

```suggestion
		message := fmt.Sprintf(`:bar_chart: Welcome to your weekly CI report for week ending: %s
```

File: dev/buildchecker/main.go [hunk 5]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
...
+		Total builds: %d
```
Comment (jhchabran) [inline dev/buildchecker/main.go:415]: It probably looks nicer with a bullet list. You can use [blockkit builder](https://app.slack.com/block-kit-builder/T02FSM7DL#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Welcome%20to%20your%20weekly%20CI%20report!%5Cn%E2%80%A2%20Total%20builds:%20*666*%5Cn%E2%80%A2%20Total%20incident%20duration:%20*7h40s*%5Cn%E2%80%A2%20Total%20flakes:%20*3*%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22image%22,%22image_url%22:%22https://i.imgur.com/Z8hRUOA.jpg%22,%22alt_text%22:%22inspiration%22%7D%5D%7D) to iterate quickly :)

File: dev/buildchecker/main.go [hunk 6]
```diff
@@ -379,10 +383,46 @@ func cmdHistory(ctx context.Context, flags *Flags, historyFlags *cmdHistoryFlags
...
+		Total incident duration: %s
+		Total flakes: %d÷
+
+		For more information, view the dashboards in [OKAYHQ](https://app.okayhq.com/dashboards/3856903d-33ea-4d60-9719-68fec0eb4313/build-stats-kpis).
+		`, historyFlags.createdToDate, totalBuilds, minutesToHours(totalTime), totalFlakes)
```
Comment (jhchabran) [inline dev/buildchecker/main.go:420]: If you're doing the computation with `time.Duration` similarly to how it's done in https://github.com/sourcegraph/sourcegraph/pull/36087/files#diff-12b4a185fe432fbafb470753ce471b501931a9d122c0c0bf2ed967d27f84006bR366, you can simply sum the durations and converting the duration into a string will give you a nice `XmYs` output without having you do handle it manually.

File: dev/buildchecker/main.go (hunk 2 above)
Comment (jhchabran) [reply dev/buildchecker/main.go:392]: The one thing that gophers really agree on, a new line in between two `}` should not be there :P

**Review by jhchabran — COMMENTED** [review]
It works perfectly, made a few minor comments 😊

File: dev/buildchecker/main.go (hunk 2 above)
Comment (davejrt) [reply dev/buildchecker/main.go:392]: https://imgflip.com/i/6hl4n1

**Review by jhchabran — APPROVED** [review]

**Pull request merged**